algorithms/src/Kokkos_Random.hpp

```diff
@@ -674,13 +674,13 @@ struct Random_UniqueIndex<Kokkos::Experimental::SYCL> {
   using locks_view_type = View<int*, Kokkos::Experimental::SYCL>;
   KOKKOS_FUNCTION
   static int get_state_idx(const locks_view_type& locks_) {
-    // FIXME_SYCL get an index that is specific to each thread.
-    // Starting from 0 for every thread is not incorrect but
-    // will lead to a lot of contention and loop iterations.
+#ifdef KOKKOS_ARCH_INTEL_GEN
```

Comment:
Is it not still a FIXME?

Reply:
I don't consider it a `FIXME` anymore. The (main) purpose of the `SYCL` backend is to run on `Intel` GPUs so I don't think it's a problem if we can't get access to the intrinsic for other (`SYCL`) devices (or if we don't tell `Kokkos` that we want to optimize for `Intel` GPUs).